pkg/apis/autoscaling/types.go

```diff
@@ -202,6 +202,62 @@ type HorizontalPodAutoscalerStatus struct {
 
 	// CurrentMetrics is the last read state of the metrics used by this autoscaler.
 	CurrentMetrics []MetricStatus
+
+	// conditions is the set of conditions required for this autoscaler to scale its target,
+	// and indicates whether or not those conditions are met.
+	Conditions []HorizontalPodAutoscalerCondition
+}
+
+// ConditionStatus indicates the status of a condition (true, false, or unknown).
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition;
+// "ConditionFalse" means a resource is not in the condition; "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+// HorizontalPodAutoscalerConditionType are the valid conditions of
+// a HorizontalPodAutoscaler.
+type HorizontalPodAutoscalerConditionType string
+
+var (
+	// CanAccessScale indicates whether or not the HPA controller can access the scale endpoint
+	// of it's target.
+	CanAccessScale HorizontalPodAutoscalerConditionType = "CanAccessScale"
+	// InBackoff indicates that the HPA is current in backoff from another scale, and thus will
+	// not attempt new scaling.
+	InBackoff HorizontalPodAutoscalerConditionType = "InBackoff"
+	// CanComputeReplicas indicates whether or not the HPA controller can properly access metrics
+	// and compute replicas for the given HPA.
+	CanComputeReplicas HorizontalPodAutoscalerConditionType = "CanComputeReplicas"
+	// DesiredOutsideRange indicates that the calculated scale based on metrics would be above or
+	// below the range for the HPA, and has thus been capped.
+	DesiredOutsideRange HorizontalPodAutoscalerConditionType = "DesiredOutsideRange"
```

Comment:
- Backoff is specifically "have we scaled recently enough that we won't scale again"
- OutsideDesiredRange is an indication that the metric is bumping up against the scaling limits set (i.e. we would scale more, but we've been capped).  It should be an indication that the user probably needs to raise their scaling cap/lower their scaling floor